src/main/java/io/temporal/activity/ActivityOptions.java

```diff
@@ -185,7 +185,7 @@ public ActivityOptions validateAndBuildWithDefaults() {
           taskQueue,
           retryOptions,
           contextPropagators,
-          cancellationType);
+          cancellationType == null ? ActivityCancellationType.TRY_CANCEL : cancellationType);
```

Comment:
[Question] - Should this type of "default" be set on the Server side instead?

Reply:
This is purely client side option. It is not passed to the service anywhere.